### PR TITLE
[do not merge yet] Port to newer upower api, dh porting, deprecate old glib api

### PR DESCRIPTION
**Pull request by MerlijnWajer**
Wanted to get @spinal84 's comments on some of the work I did here for beowulf, since he's listed as maintainer now. I did the first port a while ago and completely messed it up, but debugged that today.

I will rebase some commits (the upower api change + it's fixup fixes), and properly fix the `g_type_class_add_private` in separate commits.

But here it is, at least you're aware of the changes.

### CODE REVIEW

**Comment by MerlijnWajer** [comment]
I don't know why all these comments from earlier are included in the PR. Anyway, the interesting parts are autotools porting, upower api changes with difference refcounting, and deprecated glib api (which I reverted, I want to double check those changes)

Again, only to make you aware of the changes, no need to review/merge now.

File: configure.ac [hunk 1]
```diff
@@ -0,0 +1,49 @@
+AC_INIT([Status area battery applet],
+        [1.5.0],
+        [merlijn@wizzup.org], [status-area-applet-battery])
```
Comment (MerlijnWajer) [inline configure.ac:3]: Can/will change maintainer to you here @spinal84 - just let me know.

**Comment by MerlijnWajer** [comment]
AUTHORS file should probably include several people, too.

**Comment by spinal84** [comment]
Hi, Merlijn. Thanks for your work!

Sooner or later I'll switch to Beowulf/Buster and will update the master branch to include your changes.
Probably, I will do it manually. So there's no need to do a PR's. This would be just an extra work for you now, so don't waste your time on PR's (just a tip).

Do whatever you want in maemo/beowulf branch.

I'm going to stick on ascii/stretch for a while - maybe add some Droid 4 related fixes/improvements.
I will ping you if there will be something interesting you may want to merge to your branch.

**Comment by MerlijnWajer** [comment]
We plan to switch to beowulf very soon, and I'm personally not interested in maintaining both ascii and beowulf. The history in `maemo/beowulf` is convoluted (messy) at the moment (my bad), and I'll clean that up.

Again -- the PR is not here for you to merge it, but to be aware of the work.

Is there a reason you don't want to go to beowulf? I guess you're far along with stretch, but not with buster?

**Comment by spinal84** [comment]
You don't have to maintain ascii. I don't ask you to do so.
I can support everything that I need myself.

The main reason I'm sticking on ASCII/Stretch is the stability.
I suppose there were mistakes made in process of migrating to Beowulf.
Such mistakes will make my work (on systemd) harder and will distract me.

I need something stable to rely on while I'm working.
Once I have a working systemd environment based on Stretch I may switch to Buster.

**Comment by MerlijnWajer** [comment]
Ok; I think we're going to officially switch to Beowulf today. I'm going to work on the news post.
If there is any relevant work for the droid, please, let's make sure it ends up in beowulf too.

**Pull request merged**